consumer/consumer.go

```diff
@@ -94,9 +95,15 @@ func NewWithSaramaConfig(
 		return nil, err
 	}
 
-	payloadTrackerProducer, err := producer.New(brokerCfg)
+	payloadTrackerProducer, err := producer.CreatePayloadTrackerProducer(brokerCfg)
 	if err != nil {
-		log.Error().Err(err).Msg("unable to construct producer")
+		log.Error().Err(err).Msg("unable to construct payload tracker producer")
+		return nil, err
+	}
+
+	deadLetterProducer, err := producer.CreateDeadLetterProducer(brokerCfg)
```

Comment:
Also here. I think it is more common to use `New` instead of `Create` for instantiating new instances for a given type. Maybe I'm wrong